src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -716,11 +716,16 @@ private String retrieveGCPSnapshotRegion(SnapshotCacheResult cachedSnapshot, FSF
     if (cachedSnapshot.isSelfHosted) {
       // Authorize using the dataset's service account...
       Storage storage = gcsProjectFactory.getStorage(cachedSnapshot.datasetProjectId);
-      Bucket bucket =
-          storage.get(
-              GcsUriUtils.parseBlobUri(fsFile.getCloudPath()).getBucket(),
-              // ...but bill to the snapshot's project
-              BucketGetOption.userProject(cachedSnapshot.googleProjectId));
+      // ...but bill to the snapshot's project.
+      BucketGetOption userProject = BucketGetOption.userProject(cachedSnapshot.googleProjectId);
+      String cloudPath = fsFile.getCloudPath();
+      Bucket bucket = storage.get(GcsUriUtils.parseBlobUri(cloudPath).getBucket(), userProject);
+      if (bucket == null) {
+        throw new DrsObjectNotFoundException(
+            "GCS bucket from %s not found (diagnostics returned in error details)"
+                .formatted(cloudPath),
+            List.of(cachedSnapshot.toString(), fsFile.toString()));
```

Comment:
Why are some data provided in the initial error message and some are appended at the end? Could all the data be provided in the error message? My understanding was `causes` is used in the case where we want to collected the chained exception causes and roll them up to the top level message.

Reply:
We haven't been perfectly consistent here, but the way that this exception will ultimately be massaged into a response to the user treats any optional supplemental elements as "error details":

https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/resources/api/data-repository-openapi.yaml#L6770-L6777

Because the stringified `cachedSnapshot` and `fsFile` are both large and complex, I believe that embedding them in the top-level error message will make it harder to diagnose the issue.  This is something that could be easily changed in the future if devs and users preferred.